src/core/api/link_raw.hpp

```diff
@@ -53,69 +53,195 @@ class LinkRaw
+     * @param[in]  aError    OT_ERROR_NONE when successfully received a frame, OT_ERROR_ABORT when reception
+     *                       was aborted and a frame was not received, OT_ERROR_NO_BUFS when a frame could not be
+     *                       received due to lack of rx buffer space.
+     *
      */
     void InvokeReceiveDone(otRadioFrame *aFrame, otError aError);
 
     /**
      * This method starts a (single) Transmit on the link-layer.
      *
+     * @param[in]  aFrame               A pointer to the frame that was transmitted.
+     * @param[in]  aCallback            A pointer to a function called on completion of the transmission.
+     *
+     * @retval OT_ERROR_NONE            Successfully transitioned to Transmit.
+     * @retval OT_ERROR_INVALID_STATE   The radio was not in the Receive state.
+     *
      */
     otError Transmit(otRadioFrame *aFrame, otLinkRawTransmitDone aCallback);
 
     /**
      * This method invokes the mTransmitDoneCallback, if set.
      *
+     * @param[in]  aFrame     A pointer to the frame that was transmitted.
+     * @param[in]  aAckFrame  A pointer to the ACK frame, NULL if no ACK was received.
+     * @param[in]  aError     OT_ERROR_NONE when the frame was transmitted, OT_ERROR_NO_ACK when the frame was
+     *                        transmitted but no ACK was received, OT_ERROR_CHANNEL_ACCESS_FAILURE when the transmission
+     *                        could not take place due to activity on the channel, OT_ERROR_ABORT when transmission was
+     *                        aborted for other reasons.
+     *
      */
     void InvokeTransmitDone(otRadioFrame *aFrame, otRadioFrame *aAckFrame, otError aError);
 
     /**
      * This method starts a (single) Enery Scan on the link-layer.
      *
+     * @param[in]  aScanChannel     The channel to perform the energy scan on.
+     * @param[in]  aScanDuration    The duration, in milliseconds, for the channel to be scanned.
+     * @param[in]  aCallback        A pointer to a function called on completion of a scanned channel.
+     *
+     * @retval OT_ERROR_NONE             Successfully started scanning the channel.
+     * @retval OT_ERROR_NOT_IMPLEMENTED  The radio doesn't support energy scanning.
+     * @retval OT_ERROR_INVALID_STATE    If the raw link-layer isn't enabled.
+     *
      */
     otError EnergyScan(uint8_t aScanChannel, uint16_t aScanDuration, otLinkRawEnergyScanDone aCallback);
 
     /**
      * This method invokes the mEnergyScanDoneCallback, if set.
      *
+     * @param[in]   aEnergyScanMaxRssi  The max RSSI for energy scan.
+     *
      */
     void InvokeEnergyScanDone(int8_t aEnergyScanMaxRssi);
 
+    /**
+     * This method is called when the transmission has started.
+     *
+     * @param[in]  aFrame     A pointer to the frame that is being transmitted.
+     *
+     */
+    void TransmitStarted(otRadioFrame *aFrame);
+
+#if OPENTHREAD_RAW
```

Comment:
Why are we wrapping the methods below with `OPENTHREAD_RAW`?  If this class is included, shouldn't the functionality below also be available?